### ISSUE TITLE
Process some Picture will panic

### ISSUE THREAD

**Issue by DobyAsa** [gen2brain/jpegli#2]
My code:
```go
package main

import (
	"os"

	"github.com/gen2brain/jpegli"
)

func main() {
	file, err := os.Open("testdata/cmyk.jpg")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	img, err := jpegli.Decode(file)
	if err != nil {
		panic(err)
	}

	outputFile, err := os.Create("output.jpg")
	if err != nil {
		panic(err)
	}

	err = jpegli.Encode(outputFile, img, &jpegli.EncodingOptions{
		Quality: 80,
	})
	if err != nil {
		panic(err)
	}

	outputFile.Close()
}
```
Some picture like your testdata/cmyk.jpg, it works.
But for some picture i tried it doesn't work.
And will show the error code.
Thanks for your hard work to build the library, I did not know about wasm, so I want to ask why ?
```
panic: decode: wasm error: invalid table access
wasm stack trace:
        .$12(i32)
        .$5(i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32) i32

goroutine 1 [running]:
main.main()
        /home/ubuntu/xxxxxxxx/jpegli_test/main.go:19 +0x19c
exit status 2
```

**Comment by DobyAsa** [comment]
I just saw this [project](https://github.com/ngmisl/go-jpegli) and tried to read the image with image/* decode and it seems to work without problems.
But even though it doesn't report an error, it seems like the compression of the images is faulty, I'll put up my test images tomorrow if I have time as well.

**Comment by gen2brain** [comment]
Can you send some problematic images, or use `identify` on them, and share some info, i.e. is the resolution some weird one, what colorspace, is it CMYK as in your example, etc?

**Comment by DobyAsa** [comment]
> Can you send some problematic images, or use `identify` on them, and share some info, i.e. is the resolution some weird one, what colorspace, is it CMYK as in your example, etc?

The image info:
<img width="377" alt="截屏2024-04-10 08 38 03" src="https://github.com/gen2brain/jpegli/assets/73781773/7146eb1f-ac5a-43af-a426-6b0802a9aad7">
The image:
![default_banner](https://github.com/gen2brain/jpegli/assets/73781773/b6756bc1-a9dd-4ce0-92af-ed477e722b2b)

**Comment by gen2brain** [comment]
This should be fixed in https://github.com/gen2brain/jpegli/commit/682c6cfa9c05003ef6e6c852debc3dbea7bd673c. Please test.

**Comment by DobyAsa** [comment]
> This should be fixed in [682c6cf](https://github.com/gen2brain/jpegli/commit/682c6cfa9c05003ef6e6c852debc3dbea7bd673c). Please test.

Thanks for your fix !!!
Now the program doesn't crash, but there's a problem with the result after compression.
```go
package main

import (
	"os"

	"github.com/gen2brain/jpegli"
)

func main() {
	filepath := os.Args[1]
	file, err := os.Open(filepath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	// img, err := jpeg.Decode(file)
	img, err := jpegli.Decode(file)
	if err != nil {
		panic(err)
	}

	outputFile, err := os.Create("output.jpg")
	if err != nil {
		panic(err)
	}

	err = jpegli.Encode(outputFile, img, &jpegli.EncodingOptions{
		Quality: 100,
	})
	if err != nil {
		panic(err)
	}

	outputFile.Close()
}
```
The output:
![output](https://github.com/gen2brain/jpegli/assets/73781773/c661a0b5-90d7-4c6a-8226-37c5c15dfa3e)
<img width="377" alt="截屏2024-04-10 16 32 17" src="https://github.com/gen2brain/jpegli/assets/73781773/016bb981-71d9-4112-9393-dad8768af5c5">

**Comment by gen2brain** [comment]
Fixed in https://github.com/gen2brain/jpegli/commit/d85bf30cf07512896ae65e03651905bd762c3148.
